### PR TITLE
Update libstorage && move m_name2Table to MemoryTableFctory

### PR DESCRIPTION
**Pull request by bxq2011hust**
- move 1.5 update of libstorage
- move m_name2Table to MemoryTableFctory

### CODE REVIEW

File: libstorage/LevelDBStorage.cpp [hunk 1]
```diff
@@ -114,22 +114,21 @@ size_t LevelDBStorage::commit(
 
                 for (size_t i = 0; i < dataIt.second->size(); ++i)
                 {
-                    if (dataIt.second->get(i)->dirty())
```
Comment (wheatli) [inline libstorage/LevelDBStorage.cpp]: why can we commit all data even undirty

File: libstorage/MinerPrecompiled.cpp [hunk 2]
```diff
@@ -82,13 +93,21 @@ bytes MinerPrecompiled::call(ExecutiveContext::Ptr context, bytesConstRef param)
...
             if (entries->size() == 0u)
+            {
+                LOG(DEBUG) << "MinerPrecompiled remove node not in _sys_miners_, nodeID : "
+                           << nodeID;
+                table->insert(MINER_TYPE_OBSERVER, entry);
                 break;
-            table->remove(key, condition);
+            }
+            table->update(MINER_TYPE_MINER, entry, condition);
             LOG(DEBUG) << "MinerPrecompiled remove miner nodeID : " << nodeID;
```
Comment (wheatli) [inline libstorage/MinerPrecompiled.cpp]: shall we need to break here?

File: libstorage/TableFactoryPrecompiled.cpp [hunk 3]
```diff
@@ -102,35 +107,36 @@ bytes TableFactoryPrecompiled::call(std::shared_ptr<ExecutiveContext> context, b
...
-    if (it == m_name2Table.end())
+    if (m_sysTables.end() == find(m_sysTables.begin(), m_sysTables.end(), _tableName))
+        return Address();
+
+    auto address = m_MemoryTableFactory->getTable(_tableName);
+    if (address == Address())
     {
         dev::storage::Table::Ptr table = m_MemoryTableFactory->openTable(
-            context->blockInfo().hash, context->blockInfo().number.convert_to<int>(), tableName);
+            context->blockInfo().hash, context->blockInfo().number.convert_to<int>(), _tableName);
```
Comment (wheatli) [inline libstorage/TableFactoryPrecompiled.cpp]: converting block number to int will lose precision

File: libstorage/TableFactoryPrecompiled.cpp [hunk 4]
```diff
@@ -33,6 +32,11 @@ using namespace dev;
 using namespace dev::blockverifier;
 using namespace std;
 
+TableFactoryPrecompiled::TableFactoryPrecompiled()
+{
+    m_sysTables.push_back("_sys_tables_");
```
Comment (wheatli) [inline libstorage/TableFactoryPrecompiled.cpp]: how to adapt the multipul ledgers?

File: libstorage/LevelDBStorage.cpp (hunk 1 above)
Comment (bxq2011hust) [reply libstorage/LevelDBStorage.cpp]: LevelDB use KV. every time it changes, need write all item to leveldb.The key is `std::string entryKey = table + "_" + key;`, Valuse is `Json::Value valueJson;`.
https://github.com/FISCO-BCOS/lab-bcos/blob/bbd04b1fc406b291f314bf1a37481dfeafa08331/libstorage/LevelDBStorage.cpp#L47-L85

File: libstorage/TableFactoryPrecompiled.cpp (hunk 3 above)
Comment (wheatli) [reply libstorage/TableFactoryPrecompiled.cpp]: I think you'd better change int to int64, to prevent any mistakes

Comment (bxq2011hust) [reply libstorage/TableFactoryPrecompiled.cpp]: ok!

**Pull request merged**